### ISSUE TITLE
replaceState should trigger subscribe callbacks

### ISSUE THREAD

**Issue by miaulightouch** [vuejs/vuex#1316]
### Version
3.0.1

### Reproduction link
[https://codepen.io/anon/pen/zajRgg?editors=1010](https://codepen.io/anon/pen/zajRgg?editors=1010)

### Steps to reproduce
1. `subscribe` a callback
2. `replaceState` with any object, and `subscribe` wouldn't be triggered

### What is expected?
`subscribe` should be triggered

### What is actually happening?
it's not.

---
if `replaceState` won't fire `subscribe` callback, vue-devtools would not work properly.

ref: vuejs/vue-devtools#690

**Comment by rulrok** [comment]
I agree that we need a way of getting informed when the state has been replaced. 

The lack of this capability is making life difficult with `vuex-persisted` plugin on production mode (i.e, no `scrict` mode).

**Comment by gitlab-winnie** [comment]
Is this a duplicate of https://github.com/vuejs/vuex/issues/1448 (or the other way round)?

**Comment by nek4life** [comment]
Any movement on this? It's a bit confusing in the dev tools when calling replaceState and they do not update correctly. I looked in the devtools repo and saw this issue https://github.com/vuejs/vue-devtools/issues/690

**Comment by MarcelFinkbeiner** [comment]
any updates on this?
